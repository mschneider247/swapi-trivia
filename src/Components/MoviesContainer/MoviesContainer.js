import React from 'react';
import Movie from '../Movie/Movie';
import Profile from '../Profile/Profile';
import './MoviesContainer.css';

const MoviesContainer = ( {movies, user} ) => {

  console.log('user', user)

    const moviesCont = movies.map((movie) => {
      return <Movie
      title={movie.title}
      episode_id={movie.episode_id}
      release_date={movie.release_date}
      id={movie.episode_id}
      key={movie.episode_id}
      characters={movie.characters}/>
    })

    const profile =
      <Profile
        name={user.name}
        quote={user.quote}
        status={user.status}
      />

  return (
    <main className="movies-container">
      {profile}
      {moviesCont}
    </main>
  )
}

export default MoviesContainer;
