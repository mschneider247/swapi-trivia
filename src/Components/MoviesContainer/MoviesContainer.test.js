import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import MoviesContainer from './MoviesContainer';

describe('MoviesContainer', () => {
  it('should match snapshot with correct data', () => {
    const mockMovies = [
      1, 2, 3
    ]
    const wrapper = shallow(
      <MoviesContainer
      movies={mockMovies}/>
    );

    expect(wrapper).toMatchSnapshot()
  });

});
