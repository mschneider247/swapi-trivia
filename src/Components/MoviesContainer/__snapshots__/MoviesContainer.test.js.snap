// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should match snapshot with correct data 1`] = `
<main
  className="movies-container"
>
  <img
    alt="loading..."
    src="https://media3.giphy.com/media/hQgJCEdGOEHa8/source.gif"
  />
</main>
`;
